ClangSharpPInvokeGenerator.Test/EnumDeclarationTest.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Threading.Tasks;
+using Xunit;
+
+namespace ClangSharpPInvokeGenerator.Test
+{
+    public sealed class EnumDeclarationTest : PInvokeGeneratorTest
+    {
+        [Fact]
+        public async Task BasicTest()
```

Comment:
These tests are very basic and the general infrastructure involves creating two temporary files on disk per test (the `PInvokeGenerateTest` class is itself `IDisposable` so xUnit should ensure they are cleaned up per test as well).

For every test, it just takes an input and generates an expected output.

It should be possible to refactor things in the `ClangSharpPInvokeGenerator` to work on `Streams` instead, in which case everything could be in memory, but that might be a good thing to do in a future PR (you likely need to provide some kind of factory so that the `multi-file` configuration works with this setup as well).